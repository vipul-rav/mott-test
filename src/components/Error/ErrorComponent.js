import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ErrorComponent extends Component {
    constructor(props) {
        super(props);
        this.state = { hasError: props.hasError || false };
    }

    componentDidCatch(error, info) {
        // Display fallback UI
        this.setState({ hasError: true });
        // eslint-disable-next-line no-console
        console.log(error);
        // eslint-disable-next-line no-console
        console.log(info);
    }
    render() {
        if (this.state.hasError) {
            return (
                <div>
                    <h1>Technical Error</h1>
                    <p>There is some Technical Error.</p>
                </div>
            );
        }
        return this.props.children;
    }
}

ErrorComponent.propTypes = {
    content: PropTypes.object.isRequired,
    children: PropTypes.any,
    hasError: PropTypes.bool
};

export { ErrorComponent };
