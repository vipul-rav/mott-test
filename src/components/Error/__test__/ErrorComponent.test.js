import React from 'react';
import { render } from '@testing-library/react';
import { ErrorComponent } from '../../Error';

const setup = (overridesProps) => {
    const props = {
        ...overridesProps
    };

    const { container, getByText } = render(
        <ErrorComponent {...props}>
            <div>test</div>
        </ErrorComponent>
    );

    return {
        props,
        container,
        getByText
    };
};

describe('Error Component', () => {
    it('should render - hasError=true', () => {
        const overrideProps = {
            hasError: true,
            content: {
                errorHeader: 'Technical Error',
                errorText: 'There are some technical error',
                ExitText: 'Exit'
            }
        };
        const { container } = setup(overrideProps);
        expect(container).toMatchSnapshot();
    });
    it('should render - hasError=false', () => {
        const overrideProps = {
            hasError: false
        };
        const { container } = setup(overrideProps);
        expect(container).toMatchSnapshot();
    });
});
