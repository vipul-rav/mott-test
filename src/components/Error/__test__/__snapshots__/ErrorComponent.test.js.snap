// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component should render - hasError=false 1`] = `
<div>
  <div>
    test
  </div>
</div>
`;

exports[`Error Component should render - hasError=true 1`] = `
<div>
  <div>
    <h1>
      Technical Error
    </h1>
    <p>
      There is some Technical Error.
    </p>
  </div>
</div>
`;
